libraries/joomla/table/table.php

```diff
@@ -1382,6 +1501,14 @@ public function reorder($where = '')
 			$query->where($where);
 		}
 
+		// Pre-processing by observers
+		$event = AbstractEvent::create('onTableBeforeReorder', [
+			'subject'	=> &$this,
+			'query'		=> &$query,
+			'where'		=> $where,
+		]);
+		$this->getDispatcher()->dispatch('onTableBeforeReorder', $event);
+
```

Comment:
I've been writing code for PHP 5.3 for too long... There were some old
versions of PHP 5.3 where objects in an array passed as an argument to a
method ended up being passed by value. Only certain builds on certain
platforms. But yeah, since our minimum requirement is PHP 5.5 (or even
newer, who knows) there is no point doing that. Fixed.